src-electron/generator/matter/controller/java/templates/helper.js

```diff
@@ -28,6 +28,7 @@ const zclUtil = require(zapPath + 'util/zcl-util.js');
 
 const characterStringTypes = ['CHAR_STRING', 'LONG_CHAR_STRING'];
 const octetStringTypes = ['OCTET_STRING', 'LONG_OCTET_STRING'];
+const basicAttributeExceptionList = ['vendor_id'];
```

Comment:
why is this an exception ... we would want a comment explaining.

I cannot figure out why vendor_id, fabric_idx and group_id are not basic and had to code this into https://github.com/project-chip/connectedhomeip/pull/25786 ....

Reply:
I think this is an existing problem may it be jinja or handlebar templates. I believe the list is a lot longer than that. ```'ActionId', 'FabricIndex', 'Percent', 'EndpointId', 'GroupId', 'VendorId', 'Percent100ths', 'ClusterId', 'AttributeId', 'FieldId', 'EventId', 'CommandId', 'TransactionId', 'DeviceTypeId', 'StatusCode', 'DataVersion', 'EventNumber', 'FabricId', 'NodeId'```